packages/docusaurus-plugin-content-blog/.npmignore

```diff
@@ -0,0 +1,10 @@
+copyUntypedFiles.js
+.tsbuildinfo
+tsconfig*
+__tests__
+
+src/**
+!src/plugin-content-blog.d.ts
+!src/types.ts
+!src/blogFrontMatter.ts
```

Comment:
This is one of the cases we should protect ourselves against

Just adding a little import can break a release for TS users, and we wouldn't notice before publish that this happened.

![image](https://user-images.githubusercontent.com/749374/146222207-92a54e26-297c-4b5d-98b9-cb48d00a2c4b.png)

Is there a way that we can protect ourselves from this kind of issue before publishing?

Reply:
Yes, next step would be to refactor this and get rid of any `import('foo.ts')` in `.d.ts` files and reverse that to importing from declaration files. But since we agreed that src should be published this is not much a problem